comictaggerlib/ui/talkeruigenerator.py

```diff
@@ -0,0 +1,233 @@
+    vspacer = QtWidgets.QSpacerItem(20, 40, QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Expanding)
+    general_layout.addItem(vspacer, 1, 0)
+    source_info["general"]["widgets"]["talker_source"] = cbx_info
+
+    source_info["general"]["tab"].setLayout(general_layout)
+    tabs.addTab(source_info["general"]["tab"], "General")
+
+    # Add source sub tabs to Comic Sources tab
+    for talker_id, talker_obj in talkers.items():
+        # Add source to general tab dropdown list
+        source_info["general"]["widgets"]["talker_source"].addItem(talker_obj.name, talker_id)
+
+        tab_name = talker_id
+        source_info[tab_name] = {"tab": QtWidgets.QWidget(), "widgets": {}}
+        layout_grid = QtWidgets.QGridLayout()
+
+        for option in config[1][f"talker_{talker_id}"][1].values():
+            current_widget = None
+            if option.action is not None and (
+                option.action is argparse.BooleanOptionalAction
+                or option.action is bool
+                or option.action == "store_true"
+                or option.action == "store_false"
+            ):
+                layout_grid, current_widget = generate_checkbox(option, layout_grid)
+                source_info[tab_name]["widgets"][option.internal_name] = current_widget
+            elif option.type is int or option.type is float:
+                layout_grid, current_widget = generate_spinbox(
+                    option, getattr(config[0], option.internal_name), layout_grid
+                )
+                source_info[tab_name]["widgets"][option.internal_name] = current_widget
+            # option.type of None should be string
+            elif option.type is None or option.type is str:
+                layout_grid, current_widget, btn = generate_textbox(option, layout_grid)
+                source_info[tab_name]["widgets"][option.internal_name] = current_widget
+
+                if option.internal_name.endswith("key"):
+                    # Attach test api function to button. A better way?
+                    api_key_btn_connect(btn, talker_id, source_info, talkers)
+                if option.internal_name.endswith("url"):
+                    # Attach test api function to button. A better way?
+                    api_url_btn_connect(btn, talker_id, source_info, talkers)
+            else:
+                logger.debug(f"Unsupported talker option found. Name: {option.internal_name} Type: {option.type}")
+
+        # Add vertical spacer
+        vspacer = QtWidgets.QSpacerItem(20, 40, QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Expanding)
+        layout_grid.addItem(vspacer, layout_grid.rowCount() + 1, 0)
+        # Display the new widgets
+        source_info[tab_name]["tab"].setLayout(layout_grid)
+
+        # Add new sub tab to Comic Source tab
+        tabs.addTab(source_info[tab_name]["tab"], talker_obj.name)
+        sources.update(source_info)
```

Comment:
You are doing something wrong if you are using update here.
update is for when you get a dictionary from somewhere else and need to add all of it's content to a dictionary you got from a different place. All of these dictionaries are created here in this function.
```pycon
>>> a = {'a': True}
>>> b = {'a': False, 'b': True}
>>> a.update(b)
>>> a
{'a': False, 'b': True}
>>> a = {'a': True}
>>> a['b'] = True
>>> a
{'a': True, 'b': True}
```

Reply:
It appears that `source_info` duplicates the end result of `sources`. `source_info` was inside the tab loop so it might have made sense then. I've removed `source_info`.